### ISSUE TITLE
Problem using on VSCode on macOS

### ISSUE THREAD

**Issue by NoLooseEnds** [nvuillam/vscode-groovy-lint#66]
I installed the extension via VSCode, but I get a strange error.

```
Start analyzing file:///Users/<user>/Downloads/testfile.groovy
GroovyLint: Started CodeNarc Server
CodeNarcServer unexpected error:
{
  "message": "Request failed with status code 500",
  "name": "Error",
  "stack": "Error: Request failed with status code 500\n    at createError (/Users/<user>/.vscode/extensions/nicolasvuillamy.vscode-groovy-lint-0.18.0/server/node_modules/axios/lib/core/createError.js:16:15)\n    at settle (/Users/<user>/.vscode/extensions/nicolasvuillamy.vscode-groovy-lint-0.18.0/server/node_modules/axios/lib/core/settle.js:17:12)\n    at IncomingMessage.handleStreamEnd (/Users/<user>/.vscode/extensions/nicolasvuillamy.vscode-groovy-lint-0.18.0/server/node_modules/axios/lib/adapters/http.js:236:11)\n    at IncomingMessage.emit (events.js:208:15)\n    at endReadableNT (_stream_readable.js:1168:12)\n    at processTicksAndRejections (internal/process/task_queues.js:77:11)",
  "config": {
    "url": "http://localhost:7484/request",
    "method": "post",
    "data": "{\"codeNarcArgs\":\"-basedir=/var/folders/l6/qsdd5bln16g983pvy6jlwcqr0000gn/T/npm-groovy-lint/codeNarcTmpDir_0.9271229852517635 -rulesetfiles=file:/var/folders/l6/qsdd5bln16g983pvy6jlwcqr0000gn/T/npm-groovy-lint/codeNarcTmpRs_0.04516897782458451.groovy -includes=**/movieformat.groovy -report=xml:/var/folders/l6/qsdd5bln16g983pvy6jlwcqr0000gn/T/npm-groovy-lint/codeNarcReportXml_0.3101359203624061.xml\",\"parse\":true,\"file\":\"/var/folders/l6/qsdd5bln16g983pvy6jlwcqr0000gn/T/npm-groovy-lint/codeNarcTmpDir_0.9271229852517635/movieformat.groovy\",\"requestKey\":\"/Users/<user>/Downloads/filebot/movieformat.groovy-none\"}",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Content-Type": "application/json;charset=utf-8",
      "User-Agent": "axios/0.19.2",
      "Content-Length": 608
    },
    "transformRequest": [
      null
    ],
    "transformResponse": [
      null
    ],
    "timeout": 600000,
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
    "maxContentLength": -1
  }
}
GroovyLint: Error running CodeNarc: 
java.lang.StringIndexOutOfBoundsException: String index out of range: 71
	at java.lang.String.substring(String.java:1963)
	at org.codehaus.groovy.runtime.StringGroovyMethods.getAt(StringGroovyMethods.java:1306)
	at org.codehaus.groovy.runtime.dgm$1177.invoke(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:247)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)
	at org.codenarc.rule.groovyism.ClosureAsLastMethodParameterAstVisitor.visitMethodCallExpression(ClosureAsLastMethodParameterRule.groovy:44)
	at org.codenarc.rule.AbstractMethodCallExpressionVisitor.visitClass(AbstractMethodCallExpressionVisitor.java:40)
	at org.codenarc.rule.AbstractAstVisitorRule.applyTo(AbstractAstVisitorRule.java:97)
	at org.codenarc.rule.AbstractRule.applyTo(AbstractRule.java:143)
	at org.codenarc.rule.Rule$applyTo.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codenarc.rule.Rule$applyTo.call(Unknown Source)
	at org.codenarc.analyzer.AbstractSourceAnalyzer.collectViolations(AbstractSourceAnalyzer.groovy:40)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:193)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:194)
	at org.codenarc.analyzer.FilesystemSourceAnalyzer.processFile(FilesystemSourceAnalyzer.groovy:114)
	at org.codenarc.analyzer.FilesystemSourceAnalyzer.this$3$processFile(FilesystemSourceAnalyzer.groovy)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:351)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:61)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:203)
	at org.codenarc.analyzer.FilesystemSourceAnalyzer$_processDirectory_closure1.doCall(FilesystemSourceAnalyzer.groovy:103)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:263)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1029)
	at groovy.lang.Closure.call(Closure.java:412)
	at groovy.lang.Closure.call(Closure.java:428)
	at org.codehaus.groovy.runtime.ResourceGroovyMethods.eachFile(ResourceGroovyMethods.java:1186)
	at org.codehaus.groovy.runtime.ResourceGroovyMethods.eachFile(ResourceGroovyMethods.java:1204)
	at org.codehaus.groovy.runtime.dgm$1014.invoke(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:247)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)
	at org.codenarc.analyzer.FilesystemSourceAnalyzer.processDirectory(FilesystemSourceAnalyzer.groovy:92)
	at org.codenarc.analyzer.FilesystemSourceAnalyzer.this$3$processDirectory(FilesystemSourceAnalyzer.groovy)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:193)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:194)
	at org.codenarc.analyzer.FilesystemSourceAnalyzer.analyze(FilesystemSourceAnalyzer.groovy:79)
	at org.codenarc.analyzer.SourceAnalyzer$analyze.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)
	at org.codenarc.CodeNarcRunner.execute(CodeNarcRunner.groovy:74)
	at org.codenarc.CodeNarcRunner$execute.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:130)
	at org.codenarc.CodeNarc.execute(CodeNarc.groovy:159)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:193)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.call(PogoMetaMethodSite.java:73)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)
	at org.codenarc.CodeNarc.main(CodeNarc.groovy:135)

VsCode Groovy Lint error: EROFS: read-only file system, open './local-storage.json'
Error: EROFS: read-only file system, open './local-storage.json'
    at Object.openSync (fs.js:447:3)
    at Object.func (electron/js2c/asar.js:140:31)
    at Object.func [as openSync] (electron/js2c/asar.js:140:31)
    at Object.writeFileSync (fs.js:1206:35)
    at Object.writeFileSync (/Users/<user>/.vscode/extensions/nicolasvuillamy.vscode-groovy-lint-0.18.0/server/node_modules/jsonfile/index.js:117:13)
    at getUuidV4 (/Users/<user>/.vscode/extensions/nicolasvuillamy.vscode-groovy-lint-0.18.0/server/node_modules/npm-groovy-lint/lib/analytics.js:134:9)
    at recordAnonymousEvent (/Users/<user>/.vscode/extensions/nicolasvuillamy.vscode-groovy-lint-0.18.0/server/node_modules/npm-groovy-lint/lib/analytics.js:19:27)
    at NpmGroovyLint.postProcess (/Users/<user>/.vscode/extensions/nicolasvuillamy.vscode-groovy-lint-0.18.0/server/node_modules/npm-groovy-lint/lib/groovy-lint.js:300:38)
    at NpmGroovyLint.run (/Users/<user>/.vscode/extensions/nicolasvuillamy.vscode-groovy-lint-0.18.0/server/node_modules/npm-groovy-lint/lib/groovy-lint.js:65:24)
    at processTicksAndRejections (internal/process/task_queues.js:85:5)
```

**Comment by nvuillam** [comment]
I think there are 2 bugs:
- error writing local storage for anonymous stats (i know how to solve it, will do asap)
- unknown CodeNarc crash: please could you send me your linted file and your .groovylintrc.js so i can reproduce it ?

**Comment by nvuillam** [comment]
I fixed the second error in v0.18.1

Waiting for your feedback about the first error :)

**Comment by NoLooseEnds** [comment]
Great, thanks! 

I have not configured a `.groovylintrc.js` – and I'm using it to write instructions for [Filebot](https://www.filebot.net/). I have no prior experience with Groovy, I just wanted a formatter and validator and found your plugin. :)

Typical file would be:
```
movies/{plex.derive 
    {" " + tags} 
    {" " + [fn.match(/(?i)some|different|words/).upper()]} 
    {" " + [allOf 
        {allOf 
            {vf} 
            {hdr}
            .join(" ")
        } 
        {source} 
        {allOf 
            {ac} 
            {channels}
            .join(" ")
        } 
        {allOf
            {vc} 
            {if (bitdepth == 10)"10bit"}
            .join(" ")
        }
        .join(" - ")]
	}
	.tail
}
```

**Comment by nvuillam** [comment]
Such code does not look to be in groovy ...
I may be wrong, but do you have a link to the doc that you used to create such file ?

Filebot scripts look more groovysh 

https://github.com/filebot/scripts

**Comment by NoLooseEnds** [comment]
It's from [this post](https://www.filebot.net/forums/viewtopic.php?t=3244), at the bottom (/path/to/MovieFormat.groovy)

**Comment by nvuillam** [comment]
The code of your post works 

![image](https://user-images.githubusercontent.com/17500430/91635416-a2295380-e9f8-11ea-9abe-851a0a772c99.png)

About your code,I think there is some error related to the plex.derive, if you remove it , npm-groovy-lint returns errors and do not crash

**Comment by NoLooseEnds** [comment]
Ok, thanks for taking the time to look into to this.
